### PR TITLE
interrupt `input` command with Ctrl + C

### PR DESCRIPTION
**Pull request by ryand67**
# Description

GH Issue: [#8074](https://github.com/nushell/nushell/issues/8074)

Adding SIGINT Ctrl + C functionality to *most* of the `input` command. `input` with the flags `--suppress-output` and `--bytes-until` are executed in raw mode which allows a ^C signal to be caught as it's input, but an "optionless" call of `input` is using a standard `read_line()` method that takes a ^C input and clears the buffer of any data collected before the ^C. I'm more than happy to alter the "optionless" call but putting it in raw mode hides the user input, so at the moment that's unchanged.

# User-Facing Changes

Now ^C anywhere in the stdin stream will cause the command to SIGINT (as long as the command isn't optionless), rather than being read in as part of the input.
```
/home/rdevenney/projects/open_source/nushell〉input --suppress-output
Error: nu::shell::io_error (link)

  × I/O error
  help: SIGINT
```

```
/home/rdevenney/projects/open_source/nushell〉input --bytes-until "a"
Error: nu::shell::io_error (link)

  × I/O error
  help: SIGINT

```

As mentioned above, the default input behavior is the same.

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-command/src/platform/input.rs [hunk 1]
```diff
@@ -73,6 +73,11 @@ impl Command for Input {
                     }
                     buffer.push(buf[0]);
 
+                    if buf.contains(&3) {
```
Comment (rgwood) [inline crates/nu-command/src/platform/input.rs:76]: I had to look up what `3` was [in this context](https://en.wikipedia.org/wiki/End-of-Text_character). Mind adding a comment explaining the magic number?

**Review by rgwood — APPROVED** [review]
Thanks for the PR!

I think this is a step in the right direction. If you can add an extra comment as suggested, I'd be happy to merge the PR. It's unfortunate that `input` without any flags still can't be interrupted, but that can be fixed up separately.

Entirely aside from this PR, the existing `input` implementation makes me a bit sad. I find it quite hard to read; there are 3 separate ways to get input and the control flow is unnecessarily hard to follow. Definitely a good candidate for a refactoring.

File: crates/nu-command/src/platform/input.rs (hunk 1 above)
Comment (ryand67) [reply crates/nu-command/src/platform/input.rs:77]: I could've sworn that I had, but must've undo'd one too many times! Should be added now!

**Comment by ryand67** [comment]
@rgwood I'd be more than happy to take a look at refactoring this, does seem like there's a good bit of room to make it clearer. Let me know :)

**Comment by sholderbach** [comment]
One crate that does input reading and raw mode handling would be our very own reedline which we have to build anyways. Might be interesting if we could launch it in a minimal config for `input` as well (without history etc. as baggage).

**Comment by sholderbach** [comment]
If you are OK with it I would land this as a MVP for now.

**Comment by rgwood** [comment]
Yeah, let's land this; further improvements can be done in future PRs. Thanks for the contribution @ryand67!
